detectors/spike.py

```
@@ -15,8 +16,182 @@ class SpikeDetector(StructuralDetector):
+        smoothed_signal: np.ndarray,
+        indices: np.ndarray,
+    ) -> int:
+        if self.drop_localization == "left_edge":
+            return int(indices[properties["left_edge_idx"]])
+
+        if self.drop_localization == "left_ips":
+            try:
+                left_ips_arr = peak_widths(-smoothed_signal, [peak_idx], rel_height=1.0)[2]
+                left_idx = int(np.clip(round(float(left_ips_arr[0])), 0, len(indices) - 1))
+                return int(indices[left_idx])
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)